mount/app/api/rest/v1/servers.py

```diff
@@ -33,21 +36,29 @@ async def create_server(
         status_code=status.HTTP_201_CREATED,
         headers={
             "Location": f"/v1/servers/{resp.server_id}",
+            "ETag": get_entity_tag(resp),
         },
     )
 
 
 @router.get("/v1/servers/{server_id}")
 async def get_server(
     server_id: int,
+    if_none_match: str | None = Header(None),
     ctx: RequestContext = Depends(),
 ) -> Success[Server]:
     data = await servers.fetch_one(ctx, server_id=server_id)
     if isinstance(data, ServiceError):
         return responses.failure(data, "Failed to get server")
 
+    if if_none_match and if_none_match == get_entity_tag(data):
```

Comment:
need to add support for "*" case - https://www.rfc-editor.org/rfc/rfc9110.html#section-13.1.2-10.1